### PR TITLE
Change INavigableItem to support classified presentation.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This is for the new async FindReferences work.  The new FAR presenter window allows for classified text in it.  That will allow us to actually show classified versions of symbols like we do in quick-info and other locations.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tagging @paulvanbrenk @jramsay 

Paul/Jason.  This will affect you, but it will be very easy to incorporate.

All you have to do is the following:

```c#
    internal class NavigableItem : INavigableItem
    {
        public bool DisplayFileLocation => !ChildItems.Any();
        public ImmutableArray<TaggedText> DisplayTaggedParts{ get; }
        public Document Document { get; }
        public Glyph Glyph { get; }
        public TextSpan SourceSpan { get; }
        public ImmutableArray<INavigableItem> ChildItems { get; internal set; }

        public NavigableItem(string displayString, Document document, Glyph glyph, TextSpan sourceSpan)
        {
            if (displayString != null) { DisplayTaggedParts = ImmutableArray.Create(new TaggedText(TextTag.Text, displayString); }
            this.Document = document;
            this.Glyph = glyph;
            this.SourceSpan = sourceSpan;
            this.ChildItems = ImmutableArray<INavigableItem>.Empty;
        }
    }
```

In the future, if you want to show classified text in the new FAR window, you can just update your shim layer to return the appropriate data.

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-ide Plx review.

**Comment by jramsay** [comment]
👍 Thanks for including the TS sample above

**Comment by CyrusNajmabadi** [comment]
@DustinCampbell Can you take a quick look?

**Comment by CyrusNajmabadi** [comment]
Ping @dotnet/roslyn-ide Review plz.

**Comment by DustinCampbell** [comment]
:+1:

**Comment by CyrusNajmabadi** [comment]
Not yet.

File: src/Features/Core/Portable/Common/TaggedText.cs [hunk 1]
```diff
@@ -36,13 +39,29 @@ public TaggedText(string tag, string text)
...
+        {
+            return displayParts.Select(d =>
+                new TaggedText(SymbolDisplayPartKindTags.GetTag(d.Kind), d.ToString())).ToImmutableArray();
+        }
+
+        public static string JoinText(this ImmutableArray<TaggedText> values)
+        {
+            return values.IsDefault
+                ? null
+                : string.Join("", values.Select(t => t.Text));
```
Comment (Pilchie) [inline src/Features/Core/Portable/Common/TaggedText.cs]: Do the elements already contain whitespace where necessary?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Common/TaggedText.cs]: Yup.  Whitespace is in the parts.

**Pull request merged**